src/mlpack/core/util/using.hpp

```diff
@@ -75,6 +77,61 @@ namespace mlpack {
 
 #endif
 
+#if (ARMA_VERSION_MAJOR >= 12)
+  // By default, assume that we are using an Armadillo object.
+    template<typename MatType>
+    struct GetFillType
+    {
+      static constexpr decltype(arma::fill::none) none = arma::fill::none;
+      static constexpr decltype(arma::fill::zeros) zeros = arma::fill::zeros;
+      static constexpr decltype(arma::fill::ones) ones = arma::fill::ones;
+      static constexpr decltype(arma::fill::randu) randu = arma::fill::randu;
+      static constexpr decltype(arma::fill::randn) randn = arma::fill::randn;
+    };
+
+  #ifdef MLPACK_HAS_COOT
+    // If the matrix type is a Bandicoot type, use Bandicoot fill objects instead.
+    template<typename MatType,
+             typename = typename std::enable_if<is_coot_type<MatType>::value>::type*>
+    struct GetFillType
+    {
+      static constexpr decltype(coot::fill::none) none = coot::fill::none;
+      static constexpr decltype(coot::fill::zeros) zeros = coot::fill::zeros;
+      static constexpr decltype(coot::fill::ones) ones = coot::fill::ones;
+      static constexpr decltype(coot::fill::randu) randu = coot::fill::randu;
+      static constexpr decltype(coot::fill::randn) randn = coot::fill::randn;
```

Comment:
```suggestion
      static constexpr decltype(coot::fill::none) none;
      static constexpr decltype(coot::fill::zeros) zeros;
      static constexpr decltype(coot::fill::ones) ones;
      static constexpr decltype(coot::fill::randu) randu;
      static constexpr decltype(coot::fill::randn) randn;
```
I think that this should fix the compilation errors---since the `fill::` types depend on types and not actual instances of those types, I don't think it's necessary to actually make the objects explicitly equal to the ones held internally in Armadillo or Bandicoot.